app/models/spree/subscription.rb

```diff
@@ -127,6 +127,23 @@ def send_prior_notification
       end
     end
 
+    def cancel_stripe_subscription
+      raise "Stripe subscription ID not present" unless stripe_subscription_id.present?
+  
+      stripe_subscription = Stripe::Subscription.retrieve(stripe_subscription_id)
+      raise "Stripe subscription not found" unless stripe_subscription
+  
+      stripe_subscription.cancel
+    rescue Stripe::StripeError => e
+      raise "Stripe cancellation error: #{e.message}"
+    end
+  
+    def cancel_local_subscription
+      cancel
```

Comment:
how are you sure that if it fails it will raise exception?

Reply:
i have tested it will share screenshots if you want

Reply:
how did you test please tell me the process